libs/core/src/lib/popover/popover-mobile/popover-mobile.component.ts

```diff
@@ -0,0 +1,116 @@
+import {
+    ChangeDetectionStrategy,
+    ChangeDetectorRef,
+    Component,
+    ElementRef,
+    Inject,
+    OnDestroy,
+    OnInit,
+    Optional,
+    TemplateRef,
+    ViewChild,
+    ViewEncapsulation,
+} from '@angular/core';
+import { Subscription } from 'rxjs';
+import {
+    MobileModeBase,
+    MOBILE_MODE_CONFIG,
+    MobileModeControl,
+    MobileModeConfigToken
+} from '@fundamental-ngx/core/mobile-mode';
+import { PopoverInterface, POPOVER_COMPONENT } from '../popover.interface';
+import { takeUntil } from 'rxjs/operators';
+import { DialogService } from '@fundamental-ngx/core/dialog';
+
+@Component({
+    selector: 'fd-popover-mobile',
+    templateUrl: './popover-mobile.component.html',
+    styleUrls: ['./popover-mobile.component.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    encapsulation: ViewEncapsulation.None,
+})
+export class PopoverMobileComponent extends MobileModeBase<PopoverInterface> implements OnInit, OnDestroy {
+    /** @hidden
+     * from mobile class can not prefix _,
+     * to avoid build issues
+    */
+    childContent: TemplateRef<any> = undefined;
+
+    /** @hidden */
+    @ViewChild('dialogTemplate')
+    _dialogTemplate: TemplateRef<any>;
+
+    /** Current popover title */
+    title: string;
+
+    /** Dialog body content */
+    viewBody: TemplateRef<any>;
+    /** Dialog footer content */
+    viewFooter: TemplateRef<any>;
+
+    /** @hidden */
+    private _subscriptions = new Subscription();
+
+    constructor(
+        elementRef: ElementRef,
+        dialogService: DialogService,
+        private _changeDetectorref: ChangeDetectorRef,
+        @Inject(POPOVER_COMPONENT) _popoverComponent: PopoverInterface,
+        @Optional() @Inject(MOBILE_MODE_CONFIG) mobileModes: MobileModeConfigToken[]
+    ) {
+        super(elementRef, dialogService, _popoverComponent, MobileModeControl.POPOVER, mobileModes);
+    }
+
+    /** @hidden */
+    ngOnInit(): void {
+        this._listenOnPopoverOpenChange();
+
+        this.title = this.mobileConfig.title || '';
+        this.viewBody = (<any> this.childContent).popoverBodyContentTemplate;
+        this.viewFooter = (<any> this.childContent).popoverFooterContentTemplate;
```

Comment:
it looks like `this.childContent` is always undefined. If not then it's not clear from where this value is getting set up.

Reply:
from `popover.component.ts`, where mobile component is created:
![image](https://user-images.githubusercontent.com/3863691/123450667-081d4200-d5ee-11eb-940f-5f369f924c32.png)

Reply:
what magic, it's so not obvious.
in the begining we have declaration `childContent: TemplateRef<any> = undefined;` and here we are saying that it is a `TemplateRef`.
Can we drop `childContent: TemplateRef<any> = undefined;` on the line 37 since it doesn't make sense?

Reply:
+1 for Dima's comment

Reply:
@dimamarksman done, please check